### ISSUE TITLE
filters doesn't work on windows systems due to wrong path separator

### ISSUE THREAD

**Issue by mtestrot** [ivogabe/gulp-typescript#156]
I tried to use  a filter like:

```javascript
var tsProject = ts.createProject({
    typescript: require('typescript'),
    sortOutput: true,
    noImplicitAny: true,
    noEmitOnError: true,
});

gulp.task('ts', function() {
    var tsResult = gulp.src(['content/*.ts'])
        .on('error', function(err) {
            console.error('Error', err.message);
        })
        .pipe(sourcemaps.init())
        .pipe(ts(tsProject), undefined, ts.reporter.longReporter());
    return merge(
       tsResult.js
           .pipe(ts.filter(tsProject, { referencedFrom: ['MainModule.ts'] }))
           .pipe(concat('MainModule.js'))
           .pipe(sourcemaps.write('./', { includeContent: false }))
           .pipe(gulp.dest('content/')),
       tsResult.js
           .pipe(ts.filter(tsProject, { referencedFrom: ['SubModule.ts'] }))
           .pipe(concat('SubModule.js'))
           .pipe(sourcemaps.write('./', { includeContent: false }))
           .pipe(gulp.dest('content/'))
   );
});
```

but get the following error:
```
gulp-typescript: Could not find file C:\workspace\Apps\MyApp\Module\content\windowModule.js. Make sure you don't rename a file before you pass it to ts.filter()
gulp-typescript: Could not find file C:\workspace\Apps\MyApp\Module\content\windowModule.js. Make sure you don't rename a file before you pass it to ts.filter()
gulp-typescript: Could not find file C:\workspace\Apps\MyApp\Module\content\MainModule.js. Make sure you don't rename a file before you pass it to ts.filter()
gulp-typescript: Could not find file C:\workspace\Apps\MyApp\Module\content\MainModule.js. Make sure you don't rename a file before you pass it to ts.filter()
gulp-typescript: Could not find file C:\workspace\Apps\MyApp\Module\content\SubModule.js. Make sure you don't rename a file before you pass it to ts.filter()
gulp-typescript: Could not find file C:\workspace\Apps\MyApp\Module\content\SubModule.js. Make sure you don't rename a file before you pass it to ts.filter()
```

I think the reason for this is that the files stored in this.project.output.files are stored with slashes and not with windows like backslahes.

I fixed the problem by adding a line to filter.js
```javascript
    Filter.prototype.match = function (fileName) {
        var fileNameExtensionless = utils.splitExtension(fileName)[0];
        // replace backslash with slash
		fileNameExtensionless = fileNameExtensionless.replace(/\\/g, "\/");
        var outputFile = this.project.output.files[fileNameExtensionless];
        if (!outputFile) {
            console.log('gulp-typescript: Could not find file ' + fileName + '. Make sure you don\'t rename a file before you pass it to ts.filter()');
            return false;
        }
```

Could you please provide a release with this fix?

Thanks a lot.

**Comment by ivogabe** [comment]
Thanks for reporting, fixed. I'll release a new version soon.

**Comment by mtestrot** [comment]
Thanks.

**Comment by djflex68** [comment]
The sames problem occurs while trying to use reporters under Windows.

**Comment by ivogabe** [comment]
@djflex68 Can you give more information about your issue?

**Comment by djflex68** [comment]
Yes, my gulp task is looking like this : 
```   
var tsProject = ts.createProject({
    target: "ES5",
    module: "amd",
    noExternalResolve: true
});

var tsResult = gulp.src(
        ['./**/*.ts',
         '!./node_modules/**', '!./node-build-tasks/**'])
                       .pipe(sourcemaps.init())// This means sourcemaps will be generated 
                       .pipe(ts(tsProject), {}, reporters('gulp-typescript'));
```

Which causes the same error as described by @mtestrot

**Comment by ivogabe** [comment]
@djflex68 That should also be fixed with the fix for the original issue. In version 2.7.7 or earlier replacing `{}` with `undefined` would fix your issue, in 2.7.8 both will work.
